File: project-euler.java/src/main/java/euler/solutions/_2.java
package euler.solutions;

import euler.common.BaseSolution;

public class _2 extends BaseSolution {
    @Override
    public long solve() {
        int  a   = 1;
        int  b   = 2;
        long sum = 0;

        while (b < 4_000_000) {
            if (b % 2 == 0) {
                sum += b;
            }
            int tmp = a + b;
            a = b;
            b = tmp;
        }

        return sum;
    }
}

File: project-euler.go/solutions/_2/2.go
package _2

func Solve() int64 {
	a := 1
	b := 2
	var sum int64 = 0

	for b < 4000000 {
		if b % 2 == 0 {
			sum += int64(b)
		}
		tmp := a + b
		a = b
		b = tmp
	}

	return sum
}
